### ISSUE TITLE
Separate repository for ratified Arrowhead documents

### ISSUE THREAD

**Issue by emanuelpalm** [eclipse-arrowhead/roadmap#26]
I propose that a new repository be created under the "eclipse-arrowhead" GitHub organization that gathers documentation, primarily for the project in general, but potentially also for specific core systems, that has been reviewed and ratified by the Eclipse committers.

The point is to create a single source of authoritative documentation that advanced learners and implementers can consult. Making pull requests to this repository would be an excellent way of discussing new formal documents, such as the [X.509 Profile](https://github.com/arrowhead-f/x509-profile) I just started working on. If we also gather core system documentation there, it could be a great place to propose new core systems or discuss general core system changes not directly related to implementation.

I propose that the new repository be named "http:///github.com/eclipse-arrowhead/documentation".

@jerkerdelsing @tsvetlin

**Comment by jenseliasson** [comment]
I think this is a great idea. We should also add how-tos and tutorial like examples to lower the threshold for new developers.

Using sites like Medium could be a great way to write Getting started guides, tutorials etc.

Just my cents...

**Comment by tsvetlin** [comment]
Hello! The idea is good.

@jenseliasson Using Medium to write your own tutorials is okay. Note, to read more then 5 articles you need to pay for it, hence it might not the best for newcomers, to hit a paywall on an external site.
I'd recommend the GitHub wiki, and creating YouTube videos for this purpose.

**Comment by emanuelpalm** [comment]
@jenseliasson @tsvetlin My intention was for the repository to only contain more formal documents. You need a lot of creativity to get getting-started guides or tutorials to be of decent quality. Official documentation is expected to be a bit more mundane. Perhaps we can agree that helping people get started is not something this repository would be about?

**Comment by jerkerdelsing** [comment]
@emanuelpalm Do you mean that core system documentation (SysD, ....) will constitute the documentation of such repo?

**Comment by emanuelpalm** [comment]
@jerkerdelsing That could be a possibility. I don't think it should contain SysDDs and other white-box documentation, however.

**Comment by jenseliasson** [comment]
Hi.
I agree with you. The writing style for getting-started documentation and
tutorials is quite different from official system and service
documentation.

For me it makes sense that different sites and solutions are used. Medium
will attract a lot more attention from a wider range of developers. Wiki
etc is better for more formal technical documentation.

We also have the go-to sites for development, Stack overflow and Stack
exchange. Perhaps it would be good to have a presence from Arrowhead there?
Maybe not now, but I definitely see a value of supporting external
developers in the future.

Jens

**Comment by emanuelpalm** [comment]
@jenseliasson I agree that Stack Overflow, Stack Exchange, Medium, Wikis and so on may become relevant in the future. Right now, I see official documentation as the top priority.

@jerkerdelsing Apart from SysDs, SDs, X.509 profiles, and so on, another example of a relevant official document could be a __terminology document__ (or a _Reference Model_), which, akin to the [SOA Reference Model](http://docs.oasis-open.org/soa-rm/v1.0/soa-rm.pdf), would define all terms related to Arrowhead and describe how those interrelate. It should also clarify the relation between Arrowhead and the said SOA document, the RAMI 4.0 Reference Model, and perhaps other similar documents. There has been quite a lot of semantics and modeling work done by various parties, and I definitively think that should be captured in an official document.

Another document I believe should exists is a __documentation specification__, which would describe, in detail, what is expected from Eclipse Arrowhead documentation (SysDs, SoSDs, SDs, etc.). You, @jerkerdelsing, as well as others, have already written several papers about this, but they should be compiled into a single official document that is corrected and ratified by the Arrowhead committers.

I'm sure there are more documents like this, except for SoSDs, SDs, IDDs, CPs, and so on, that could be of great value.

**Comment by emanuelpalm** [comment]
Another official document I believe should exist is a definition of __Arrowhead compliance__. The document should specify what, exactly, is required in order to claim that a given system is Arrowhead-compatible. There has been some work into dividing system into legacy, adapters, and so on. This document should outline all of that.

Such a document could be a boon to us Arrowhead consultants, as it means that we both can produce our own Arrowhead-compliant systems (which in extension is a way of saying that something is Industry 4.0-compliant) and issue Arrowhead compliance-certificates for component and product suppliers with less stakes in Arrowhead. The document should perhaps not be at the top of the documentation wish-list, however, as I envision it would, in practice, be a long list of references to other Arrowhead documents.

**Comment by lgrkvst** [comment]
...for instance third-party vendors looking to brand their software "Arrowhead Compliant", analogously with the BlueTooth stamp. In essence, the document would specify exactly what that entails.

**Comment by MatsBJohansson** [comment]
Idea for structuring the need of documentation, to be aligned with the progress in this issue. Ref Roadmap WG 2021-08-18.
[Documentation strategy - v03.pptx](https://github.com/eclipse-arrowhead/roadmap/files/7007650/Documentation.strategy.-.v03.pptx)

**Comment by emanuelpalm** [comment]
@MatsBJohansson I interpret the third page of your slides as naming various types of documentation that should exist. Perhaps you could suggest a critera for documents that should go into an official Eclipse Arrowhead documentation repository on GitHub?

**Comment by emanuelpalm** [comment]
@jerkerdelsing @lgrkvst Are we talking about product/software certificates or X.509/crypto certificates?

**Comment by lgrkvst** [comment]
I'm talking about product/software certificates.

**Comment by jerkerdelsing** [comment]
@emanuelpalm @lgrkvst In connection with the on-boarding procedure arrowhead X.509 certificates are provided to onboarded devices, sw-systems and services. To get such certificate the onboarded devices and/or sw-system should exhibit a product/software certificate to be identified as the intended product/software.

**Comment by emanuelpalm** [comment]
@jerkerdelsing, I'm sorry if I wasn't clear enough. The kind of Arrowhead certificates I was talking about above are akin to those produced by an organization like [TCO](https://tcocertified.com/). I guess that if a consultancy, like Pinterop, conducted this kind of certification, then that consultancy could issue X.509 certificates. If an on-boarding system would be configured to trust the issuing certificate of that consultancy, then on-boarding could be made very seamless indeed. I'm sure additional security checks would have to be implemented, but the idea is interesting. @mzsilak, any thoughts?
